physx/src/aggregate.rs

```diff
@@ -0,0 +1,112 @@
+use crate::{
+    actor::{Actor, ActorMap},
+    articulation_base::ArticulationBase,
+    bvh_structure::BVHStructure,
+    owner::Owner,
+    traits::Class,
+};
+
+use std::marker::PhantomData;
+
+use physx_sys::{
+    PxAggregate_addActor_mut,
+    PxAggregate_addArticulation_mut,
+    PxAggregate_getActors,
+    PxAggregate_getMaxNbActors,
+    PxAggregate_getNbActors,
+    PxAggregate_getSelfCollision,
+    PxAggregate_release_mut,
+    PxAggregate_removeActor_mut,
+    PxAggregate_removeArticulation_mut,
+    //PxAggregate_getScene_mut,
+    //PxAggregate_getConcreteTypeName,
+};
+
+use std::ptr::null;
+
+#[repr(transparent)]
+pub struct Aggregate<L, S, D, H, M> {
+    obj: physx_sys::PxAggregate,
+    phantom_user_data: PhantomData<(L, S, D, H, M)>,
+}
+
+unsafe impl<P, L, S, D, H, M> Class<P> for Aggregate<L, S, D, H, M>
+where
+    physx_sys::PxAggregate: Class<P>,
+{
+    fn as_ptr(&self) -> *const P {
+        self.obj.as_ptr()
+    }
+
+    fn as_mut_ptr(&mut self) -> *mut P {
+        self.obj.as_mut_ptr()
+    }
+}
+
+impl<L, S, D, H, M> Aggregate<L, S, D, H, M> {
+    pub(crate) unsafe fn from_raw(ptr: *mut physx_sys::PxAggregate) -> Option<Owner<Self>> {
+        Owner::from_raw(ptr as *mut Self)
+    }
+
+    pub fn add_actor(&mut self, actor: &mut impl Actor, bvh: Option<&BVHStructure>) -> bool {
+        unsafe {
+            PxAggregate_addActor_mut(
+                self.as_mut_ptr(),
+                actor.as_mut_ptr(),
+                bvh.map(Class::as_ptr).unwrap_or(null()),
```

Comment:
`unwrap_or_else` is lazy, but `null` and `null_mut` are `const fn`s so they are the eagerest of eagerly evaluated, I guess that's why clippy is okay with it.  As far as I know, `unwrap_or_else(null)` and `unwrap_or(null())` are synonymous here, since `null()` is just a fancy way of writing 0.

Reply:
Ah, I wasn't aware that they were const fns. Great! :-)